src/components/Game.jsx

```diff
@@ -38,30 +38,35 @@ class Game extends Component {
   // 2.6 Mutate the state
   // 3. render the board with updated tile so it renders as active
   handleClick(rowId, columnId) {
-    const newBoard = copyBoard(this.state.board);
-    const selected = newBoard[rowId][columnId].selected;
+    const selected = this.state.board[rowId][columnId].selected;
     let currentWord = this.state.currentWord;
     let newCurrentWordPosition = this.state.currentWordPosition.slice();
     const lastLetter = newCurrentWordPosition[newCurrentWordPosition.length - 1];
 
     if (this._tileClickedIsLastSelected(selected, rowId, columnId, lastLetter)) {
-
+      const newBoard = copyBoard(this.state.board);
```

Comment:
I think we need to copy the new board because we are modifying the selected attribute newBoard[rowId][columnId].selected = false; React recommends to modify data without mutations for performance reasons.